semantic_search.py

```diff
@@ -0,0 +1,85 @@
+import argparse
+import os
+from functools import cache
+
+import numpy as np
+import torch
+from sentence_transformers import util
+
+
+@cache
+def get_title_embeddings(filename: str) -> np.ndarray:
+    return np.load(filename + "_title_emb.npy")
+
+
+@cache
+def get_abs_embeddings(filename: str) -> np.ndarray:
+    return np.load(filename + "_abs_emb.npy")
+
+
+@cache
+def get_cat_ohe(filename: str) -> np.ndarray:
+    return np.load(filename + "_cat_ohe.npy")
+
+
+def find_row_numbers(ids: list, arxiv_ids: np.ndarray) -> np.ndarray:
+    return [np.where(arxiv_ids == id)[0][0] for id in ids]
+
+
+def top_hits(
+    title_embeddings: np.ndarray,
+    abs_embeddings: np.ndarray,
+    cat_ohe: np.ndarray,
```

Comment:
Instead of passing these as parameters, you can create a getter function to return the model and use it in the function. There are pros and cons to this approach, so happy to discuss

Reply:
Again, check the new version to see if this still applies